packages/core/admin/admin/src/content-manager/components/DynamicTable/CellContent/CellValue.js

```diff
@@ -27,13 +27,31 @@ const CellValue = ({ type, value }) => {
     });
   }
 
-  if (['float', 'integer', 'biginteger', 'decimal'].includes(type)) {
-    formattedValue = formatNumber(value);
+  if (['float', 'decimal'].includes(type)) {
+    const numberOfDecimals = getNumberOfDecimals(value);
+
+    formattedValue = formatNumber(value, {
+      minimumFractionDigits: numberOfDecimals,
+      maximumFractionDigits: numberOfDecimals,
+    });
+  }
+
+  if (['integer', 'biginteger'].includes(type)) {
+    formattedValue = formatNumber(value, { maximumFractionDigits: 0 });
   }
 
   return toString(formattedValue);
 };
 
+const getNumberOfDecimals = value => {
```

Comment:
Hi @oscarsandberg, thanks for your contribution. 
Can you move this function in a utils file please : `CellContent/utils/getNumberOfDecimals.js` and add some test in `CellContent/utils/tests/getNumberOfDecimals.test.js` please 😄 .

Reply:
Hi @HichamELBSI, thanks for the comments. I have moved the function and added som tests. Let me know if you want me to do anything else :)